### DISCUSSION TITLE
Example User function from docs not working

### DISCUSSION THREAD

**Discussion by rossoj85** [SilentVoid13/Templater#560] [Help] [answered]
Hello, 

I am trying to follow the example in the docs but it is not working 

![image](https://user-images.githubusercontent.com/25422645/154162739-73a587d6-e9d7-4883-bc08-70e4b2ac2df0.png)

![image](https://user-images.githubusercontent.com/25422645/154162766-fe325b7c-3fde-44ba-8695-5cfffec6c2be.png)

![image](https://user-images.githubusercontent.com/25422645/154162785-a4d2cbf5-dfa4-43fe-9d8e-3bdbca892ba6.png)

![image](https://user-images.githubusercontent.com/25422645/154162804-223f30f9-bf7c-42c4-8da8-b39bae80e7a9.png)

![image](https://user-images.githubusercontent.com/25422645/154162822-c9d29fd5-529a-4cbc-ba9e-f3b77ee5150a.png)

Not sure why my_script is not being recognized as a function. 

Could anyone help?

**Comment by shabegom** [comment]
Name the script and the function inside the script the same.

**Reply by rossoj85** [reply]
Don't think that this is the issue. I tried again with a new script - returnHello. As soon as I start typing the script into a template I get the message:

 "Uncaught TypeError: Cannot read property 'functions' of undefined at Documentation.get_all_functions_documentation "

If I try logging tp.user, I get an empty object. It seems that the functions written in the Scrips folder are not being picked up or read by templater.  Screens are shown below in case I am missing something. 

![image](https://user-images.githubusercontent.com/25422645/154597750-98aff26e-89f9-4ac6-9b75-966c347d4d5f.png)

![image](https://user-images.githubusercontent.com/25422645/154597712-030200f7-df38-43df-aac0-8e8ff22229dc.png)

![image](https://user-images.githubusercontent.com/25422645/154597776-d143cf6f-8c2d-4f1a-8b40-5697fb8ab461.png)

![image](https://user-images.githubusercontent.com/25422645/154597800-1067b3d7-40a5-4c65-a51d-f5775de115ff.png)

**Reply by rossoj85** [reply]
My console: 


![image](https://user-images.githubusercontent.com/25422645/154597836-9d6c70c0-1961-4cff-a014-78013b4e499e.png)

**Reply by shabegom** [reply]
~~Right below where you set the Scripts folder in the Settings it should list out all the scripts Templater has indexed. Do your scripts show up there?~~

Disregard! You solved it!

**Reply by rossoj85** [reply]
Thanks anyway!

**Comment by rossoj85** [comment, marked as answer]
I solved this.

I was writing my script in Obsidian itsself and my file was getting saved as a .md. I thought that any file with .js in its name would be recognized as a script file by obsidian

Used my code editor to save a .js file in the scripts folder. I don't see it under the file system in obsidian, but its there and working
